File: Programmers/1/155652. 둘만의 암호/둘만의 암호.cpp
#include <string>
#include <vector>
#include <unordered_set>
#include <algorithm>

using namespace std;

string solution(string s, string skip, int index) {
    string answer = "";
    unordered_set<char> skipSet(skip.begin(), skip.end());

    for (char c : s) {
        int count = 0;
        char current = c;

        while (count < index) {
            current++;
            if (current > 'z') current = 'a';

            if (skipSet.find(current) == skipSet.end()) {
                count++;
            }
        }

        answer += current;
    }
    
    return answer;
}

File: Programmers/1/155652. 둘만의 암호/둘만의 암호.java
import java.util.*;

class Solution {
    public String solution(String s, String skip, int index) {
        String answer = "";
        Set<Character> skipSet = new HashSet<>();
        
        for (char ch : skip.toCharArray()) {
            skipSet.add(ch);
        }

        StringBuilder result = new StringBuilder();

        for (char c : s.toCharArray()) {
            int count = 0;
            char current = c;

            while (count < index) {
                current++;
                if (current > 'z') current = 'a';

                if (!skipSet.contains(current)) {
                    count++;
                }
            }
            
            result.append(current);
        }

        answer = result.toString();
        
        return answer;
    }
}